Adding weather events

Daniel-Glz opened the pull request.

Daniel-Glz committed:
Adding weather events

connerdouglass commented:
Other than my one small comment, I think this looks great. Maybe @AL1L can review it also and then we'll merge

AL1L commented on src/events/types/weather/LightningStrikeEvent.ts:
```diff
@@ -0,0 +1,53 @@
+    /*
+    ---------- LightningStrike entity needs to be added -----------
+
+    public getLightning():LightningStrike {
```
You can just return an Entity for now. 

Although as the Entity type has only one extra method, I'd prefer you go ahead and make the Entity class, don't have to though.

AL1L commented on src/events/types/weather/LightningStrikeEventCause.ts:
```diff
@@ -0,0 +1,9 @@
+export enum LightningStrikeEventCause {
+    COMMAND = 'COMMAND',
```
This file needs documentation

AL1L commented on src/events/types/weather/LightningStrikeEvent.ts:
```diff
@@ -0,0 +1,53 @@
+    }
+   
+    /**
+    * Gets the cause of this lightning strike
```
Almost all of your method documentation is missing one space, the `*`s should all line up. I guess it's not that important, but would be nice to fix if you have the time

AL1L commented on src/events/types/weather/LightningStrikeEvent.ts:
```diff
@@ -0,0 +1,53 @@
+    * Gets the cause of this lightning strike
+    * @returns strike cause
+    */
+    public getCause():LightningStrikeEventCause {
```
Missing a space before the return type.

@connerdouglass I was thinking, we should add eslint and prettier

AL1L commented on src/events/types/weather/LightningStrikeEvent.ts:
```diff
@@ -0,0 +1,53 @@
+    * @returns strike cause
+    */
+    public getCause():LightningStrikeEventCause {
+        return this.toJava().getCause();
```
This should have `.getNane()` to get the enum name.

Daniel-Glz committed:
Weather events details

Daniel-Glz committed:
added getName() method on the return of getLightning method

Daniel-Glz committed:
fixed error getLightning

method getName() removed from getLightning return and added to getCause() return

Daniel-Glz committed:
added lightningStrikeEventCause documentation

Daniel-Glz committed:
Merge branch 'WeatherEvents' of https://github.com/Daniel-Glz/core into WeatherEvents

Daniel-Glz committed:
Update LightningStrikeEvent.ts

connerdouglass commented:
Today I made some changes on `master` to the entity classes. Please merge your branch with the latest changes on master, and use the `LightningStrike` entity I added there, instead of yours. Thanks!

Daniel-Glz committed:
Branch merged with the latest changes on master

Daniel-Glz committed:
Merge branch 'WeatherEvents' of https://github.com/Daniel-Glz/core into WeatherEvents

Daniel-Glz commented:
> Today I made some changes on `master` to the entity classes. Please merge your branch with the latest changes on master, and use the `LightningStrike` entity I added there, instead of yours. Thanks!

Ready, my branch is up to date with the latest changes.
